tests/unit/testplan/testing/multitest/test_result.py

```diff
@@ -43,17 +43,29 @@ def intermediary(result, description=None):
     helper(result, description=description)
 
 
+@result_mod.collect_code_context
 def test_group_marking():
     """
     Tests, at result object level, if marking works as expected.
     """
     result = result_mod.Result()
     result.equal(1, 1)
-    assert result.entries.pop().line_no == get_line_no(test_group_marking, 5)
-    helper(result)
-    assert result.entries.pop().line_no == get_line_no(helper, 1)
-    intermediary(result)
-    assert result.entries.pop().line_no == get_line_no(intermediary, 2)
+
+    # Functions to test marking, with relative position of the assertions
+    functions = [
+        (test_group_marking, 6),
+        (helper, 1),
+        (intermediary, 2),
+    ]
+    current_function = inspect.currentframe().f_code.co_name
+    for function, relative_position in functions:
+        # To prevent infinite recursion
```

Comment:
The modification to this testcase doesn't look right to me.

Let's discuss offline.

Reply:
Reverted back to the previous style as we discussed